File: backend/app/schemas/payroll.py
from pydantic import BaseModel
from datetime import datetime
from typing import List


class PayrollGenerateRequest(BaseModel):
    month: int
    year: int


class PayrollRecordResponse(BaseModel):
    staff_id: int
    total_hours: float
    salary_amount: float


class PayrollPeriodResponse(BaseModel):
    id: int
    month: int
    year: int
    is_locked: bool
    generated_at: datetime
    records: List[PayrollRecordResponse]

    class Config:
        from_attributes = True

File: frontend/src/api/payroll.ts
import { apiClient } from "./client";

export interface PayrollRecord {
  staff_id: number;
  total_hours: number;
  salary_amount: number;
}

export interface PayrollPeriod {
  id: number;
  month: number;
  year: number;
  is_locked: boolean;
  generated_at: string;
  records: PayrollRecord[];
}

export const generatePayroll = async (month: number, year: number) => {
  const response = await apiClient.post("/payroll/generate", {
    month,
    year,
  });
  return response.data;
};

export const fetchPayrollPeriods = async (): Promise<PayrollPeriod[]> => {
  const response = await apiClient.get("/payroll");
  return response.data;
};